File: tests/ctx/global_tpl__ctx2.py
# -*- coding: UTF-8 -*-

# Contemplate cached template 'global'
def __getTplClass__(Contemplate):
    # extends the main Contemplate.Template class
    class Contemplate_global_Cached__ctx2(Contemplate.Template):
        'Contemplate cached template global'
        # constructor
        def __init__(self, id=None):
            self_ = self
            super(Contemplate_global_Cached__ctx2, self_).__init__( id )
            # extend tpl assign code starts here
            
            # extend tpl assign code ends here
        # tpl-defined blocks render code starts here
        
        # tpl-defined blocks render code ends here
        # render a tpl block method
        def renderBlock(self, block, data, __i__=None):
            self_ = self
            __ctx = None
            if not __i__:
                __i__ = self_
                __ctx = Contemplate._set_ctx( self_._ctx )
            r = ''
            method = '_blockfn_' + block
            if (hasattr(self_, method) and callable(getattr(self_, method))):
                r = getattr(self_, method)(data, self_, __i__)
            elif self_._extends:
                r = self_._extends.renderBlock(block, data, __i__)
            if __ctx:  Contemplate._set_ctx( __ctx )
            return r
        # render method
        def render(self, data, __i__=None):
            self_ = self
            __ctx = None
            __p__ = ''
            if not __i__:
                __i__ = self_
                __ctx = Contemplate._set_ctx( self_._ctx )
            if self_._extends:
                __p__ = self_._extends.render(data, __i__)

            else:
                # tpl main render code starts here
                
                __p__ += '' + "\n" + '' + str(Contemplate.locale("global") ) + '' + "\n" + '' + "\n" + '' + str(Contemplate.plg_("my_plugin", Contemplate.locale("ctx") ) ) + '' + "\n" + ''
                
                # tpl main render code ends here

            if __ctx:  Contemplate._set_ctx( __ctx )
            return __p__
    return Contemplate_global_Cached__ctx2
# allow to 'import *'  from this file as a module
__all__ = ['__getTplClass__']

File: tests/ctx/global_tpl__ctx2.js

!function (root, moduleName, moduleDefinition) {
var m;
// node, CommonJS, etc..
if ( 'object' === typeof(module) && module.exports ) module.exports = moduleDefinition();
// browser and AMD, etc..
else (root[ moduleName ] = m = moduleDefinition()) && ('function' === typeof(define) && define.amd && define(moduleName,[],function(){return m;}));
}(this, 'Contemplate_global_Cached__ctx2', function( ){
"use strict";
return function( Contemplate ) {
/* Contemplate cached template 'global', constructor */
function Contemplate_global_Cached__ctx2( id )
{
    var self = this;
    Contemplate.Template.call( self, id );
    /* tpl-defined blocks render code starts here */
    
    /* tpl-defined blocks render code ends here */
    /* extend tpl assign code starts here */
    
    /* extend tpl assign code ends here */
}
/* extends main Contemplate.Template class */
Contemplate_global_Cached__ctx2.prototype = Object.create(Contemplate.Template.prototype);
/* render method */
Contemplate_global_Cached__ctx2.prototype.render = function( data, __i__ ) {
    "use strict";
    var self = this, __p__ = '', __ctx = null;
    if ( !__i__ )
    {
        __i__ = self;
        __ctx = Contemplate._set_ctx( self._ctx );
    }
    /* tpl main render code starts here */
    
    __p__ += '' + "\n" + '' + (Contemplate.locale("global")) + '' + "\n" + '' + "\n" + '' + (Contemplate.plg_("my_plugin", Contemplate.locale("ctx") )) + '' + "\n" + '';
    
    /* tpl main render code ends here */
    if ( __ctx )  Contemplate._set_ctx( __ctx );
    return __p__;
};
// export it
return Contemplate_global_Cached__ctx2;
};
});
